pkg/routeagent/controllers/route/route_test.go

```diff
@@ -0,0 +1,35 @@
+package routecontroller
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Route", func() {
+	Describe("Unit test for router.go", func() {
+		var rc RouteController
+		var testinputcidrblocks []string
+
+		BeforeEach(func() {
+			testinputcidrblocks = []string{"10.10.10.0/24", "192.168.1.0/24"}
+		})
+
+		Context("Testing for function populateCidrBlockList", func() {
+			It("should append cidrblock to subnet if subnets doesnot contain cidrblock", func() {
```

Comment:
"does not" - 2 words. I think cidrblock should be 2 words as well. How about this text: "should append cidr blocks to subnets that are not already present"

Reply:
Agree for "doesnot". I kept "cidrblock" as single so that it resembles to the var name and is easy to understand/read and relate.

Reply:
But remember in BDD, the text should read as natural language and not in terms of code constructs, at least as much as possible.

Reply:
I think for proper BDD of the form given/when/then, the "if subnets does not contain cidrblock" clause is really the 'when' part and thus should be part of the Context clause. The Describe clause should indicate the 'given' part, in this case that it's testing the populateCidrBlockList function. So to put it all together, it becomes:

    Describe("Function populateCidrBlockList", func() {
        Context("When the existing subnets does not contain an input cidr block", func() {
            It("should append the cidr block to subnets", func() {
                routeController := RouteController{subnets: []string{"192.168.1.0/24"}}
                routeController.populateCidrBlockList([]string{"10.10.10.0/24", "192.168.1.0/24"})
                want := []string{"192.168.1.0/24", "10.10.10.0/24"}
                Expect(routeController.subnets).To(Equal(want))
            })
        })

        Context("When the existing subnets does contain an input cidr block", func() {
            It("should not append the cidr block to subnets", func() {
                ...
            })
        })
    })

I employed similar methodology in https://github.com/rancher/submariner/pull/40 and I think it makes sense (I'm still getting familiar with BDD).

Also notice that I eliminated the global vars and the need for BeforeEach in the code snippet above.  I think in general the use of these makes the code a bit messier and should be avoided unless there's a compelling reason to use them, eg if you need to do expensive or lengthy setup.

Reply:
Done.